src/org/zaproxy/zap/extension/saml/SAMLInspectionResult.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Zed Attack Proxy (ZAP) and its related class files.
+ *
+ *  ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ *  Copyright 2018 The ZAP Development Team
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.saml;
+
+import org.parosproxy.paros.network.HtmlParameter;
+
+public class SAMLInspectionResult {
+
+    private boolean hasSAMLMessage;
+    private HtmlParameter evidence;
+
+    public SAMLInspectionResult(boolean hasSAMLMessage, HtmlParameter evidence) {
```

Comment:
This (a public constructor) could be changed to just require the `evidence` which already implies that it has a SAML message and since the class is immutable it can provide a single instance for when it hasn't (e.g. `SAMLInspectionResult.NOT_SAML`)